action_plugins/play_sound/__init__.py

```diff
@@ -0,0 +1,183 @@
+from xml.etree import ElementTree
+
+from PySide6 import QtCore
+from PySide6.QtCore import Property, Signal
+
+from gremlin import event_handler, util
+from gremlin.base_classes import AbstractActionData, AbstractFunctor, \
+    Value
+
+from gremlin.profile import Library
+from gremlin.types import ActionProperty, InputType, PropertyType
+
+from gremlin.ui.action_model import SequenceIndex, ActionModel
+from gremlin.error import GremlinError
+from gremlin.audio_player import AudioPlayer
+
+if TYPE_CHECKING:
+    from gremlin.ui.profile import InputItemBindingModel
+
+
+class PlaySoundFunctor(AbstractFunctor):
+
+    """Executes a Play Sound action callback."""
+
+    def __init__(self, action: PlaySoundData):
+        super().__init__(action)
+
+    def __call__(
+        self,
+        event: event_handler.Event,
+        value: Value
+    ) -> None:
+        if not self._should_execute(value):
+            return
+
+        audio_player = AudioPlayer()
+        audio_player.play(self.data.sound_filename, int(self.data.sound_volume))
+
+
+class PlaySoundModel(ActionModel):
+
+    fileChanged = Signal()
+    soundVolumeChanged = Signal()
+
+    def __init__(
+            self,
+            data: AbstractActionData,
+            binding_model: InputItemBindingModel,
+            action_index: SequenceIndex,
+            parent_index: SequenceIndex,
+            parent: QtCore.QObject
+    ):
+        super().__init__(data, binding_model, action_index, parent_index, parent)
+
+    def _qml_path_impl(self) -> str:
+        return "file:///" + QtCore.QFile(
+            "core_plugins:play_sound/PlaySoundAction.qml"
+        ).fileName()
+
+    def _icon_string_impl(self) -> str:
+        return PlaySoundData.icon
+
+    def _get_sound_filename(self) -> str:
+        return self._data.sound_filename
+
+    def _set_sound_filename(self, value: str) -> None:
+        if str(value) == self._data.sound_filename:
+            return
+        self._data.sound_filename = str(value)
+        self.fileChanged.emit()
+
+    def _get_sound_volume(self) -> int:
+        return self._data.sound_volume
+
+    def _set_sound_volume(self, value: int) -> None:
+        if value == self._data.sound_volume:
+            return
+        self._data.sound_volume = value
+        self.soundVolumeChanged.emit()
+
+    sound_filename = Property(
+        str,
+        fget=_get_sound_filename,
+        fset=_set_sound_filename,
+        notify=fileChanged
+    )
+
+    sound_volume = Property(
+        int,
+        fget=_get_sound_volume,
+        fset=_set_sound_volume,
+        notify=soundVolumeChanged
+    )
+
+
+class PlaySoundData(AbstractActionData):
+
+    """Model for the play sound action."""
+
+    version = 1
+    name = "Play Sound"
+    tag = "play-sound"
+    icon = "\U0001F39C"
+
+    functor = PlaySoundFunctor
+    model = PlaySoundModel
+
+    properties = [
+        ActionProperty.ActivateOnPress,
+        ActionProperty.AlwaysExecute
```

Comment:
Not sure it's necessary to be able to play sounds when a profile is paused, i.e. not meant to react.

Reply:
Copy-paste leftover. Will update.